krkn/scenario_plugins/lp_chaos/cnv/snapshot_disruption/snapshot_disruption_base.py

```diff
@@ -0,0 +1,281 @@
+        self.logger.info(f"Deleting VM '{self.vm_name}'...")
+        try:
+            invoke(f"oc delete vm {self.vm_name} -n {self.vm_namespace} --ignore-not-found=true")
+            run(f"oc wait --for=delete vm/{self.vm_name} -n {self.vm_namespace} --timeout=120s")
+            self.logger.info(f"VM '{self.vm_name}' deleted successfully.")
+        except Exception as e:
+            self.logger.warning(f"Failed to confirm VM deletion, it might have been already deleted: {e}")
+            
+    # -- Snapshot and Controller Operations --
+
+    def _create_snapshot(self) -> bool:
+        """Creates a VirtualMachineSnapshot resource for the test VM."""
+        try:
+            snapshot_yaml = f"""
+apiVersion: snapshot.kubevirt.io/v1alpha1
+kind: VirtualMachineSnapshot
+metadata:
+  name: {self.snapshot_name}
+  namespace: {self.vm_namespace}
+spec:
+  source:
+    apiGroup: kubevirt.io
+    kind: VirtualMachine
+    name: {self.vm_name}
+"""
+            with open(f"/tmp/{self.snapshot_name}.yaml", "w") as f:
+                f.write(snapshot_yaml)
+
+            invoke(f"oc apply -f /tmp/{self.snapshot_name}.yaml")
+            return True
+        except Exception as e:
+            log_exception(f"Error creating snapshot '{self.snapshot_name}': {e}")
+            return False
+
+    def _verify_snapshot_created(self) -> bool:
+        """Waits for the snapshot to reach the 'Ready' condition."""
+        try:
+            invoke(f"oc wait --for=condition=Ready --timeout=300s vmsnapshot/{self.snapshot_name} -n {self.vm_namespace}")
+            output = invoke(f"oc get vmsnapshot/{self.snapshot_name} -n {self.vm_namespace}")
+            self.logger.info(f"Final snapshot status:\n{output}")
+            return True
+        except Exception as e:
+            log_exception(f"Error waiting for snapshot '{self.snapshot_name}' to complete: {e}")
+            return False
+
+    def _get_snapshot_controller_pod_name(self) -> str | None:
+        """Finds the name of the active csi-snapshot-controller pod."""
+        try:
+            pod_name = invoke(
+                f"oc get pods -n {self.snapshot_controller_namespace} "
+                f"-l {self.snapshot_controller_label} "
+                "-o jsonpath='{.items[0].metadata.name}'"
+            ).strip()
+
+            if not pod_name:
+                self.logger.warning("Could not find the snapshot controller pod.")
+                return None
+            return pod_name
+        except Exception as e:
+            log_exception(f"Error finding the snapshot controller pod: {e}")
+            return None
+
+    def _kill_snapshot_controller(self) -> bool:
+        """Deletes the snapshot controller pod to simulate a failure."""
+        pod_name = self._get_snapshot_controller_pod_name()
+        if not pod_name:
+            return False
+
+        try:
+            invoke(f"oc delete pod {pod_name} -n {self.snapshot_controller_namespace}")
```

Comment:
here you should be able to use [delete_pod](https://github.com/krkn-chaos/krkn-lib/blob/3e7f7300451515f95e07ad212795e312e2e33388/src/krkn_lib/k8s/krkn_kubernetes.py#L1165 )